dev/tasks/python-wheels/osx-build.sh

```diff
@@ -191,18 +191,23 @@ function build_wheel {
            --bundle-arrow-cpp --bundle-boost --boost-namespace=arrow_boost \
            bdist_wheel
     ls -l dist/
+    popd
+    popd
+}
+
+function run_tests {
+    # Runs tests on installed distribution from an empty directory
+    python --version
 
-    # Do a test installation of the built wheel and change into another
-    # directory to ensure our import tests later on pick up the wheel and
-    # not the binaries from the build directory.
-    for wheel in dist/*.whl; do
-      pip install "$wheel"
-    done
-    mkdir -p tmp
-    pushd tmp
+    # Test optional dependencies
     python -c "import pyarrow"
+    python -c "import pyarrow.orc"
     python -c "import pyarrow.parquet"
-    popd
+    python -c "import pyarrow.plasma"
+
+    pip install pytest
+    py.test -v -r sxX --durations=15 --parquet ${VIRTUAL_ENV}/lib/*/site-packages/pyarrow
```

Comment:
Not sure we need the verbosity (`-v`). Also should install `pytest-faulthandler` to get a bit more context with crashes.